app/models/category.rb

```diff
@@ -259,7 +256,9 @@ def self.preload_user_fields!(guardian, categories)
       category.permission = CategoryGroup.permission_types[:full] if guardian.is_admin? ||
         allowed_topic_create_ids&.include?(category[:id])
 
-      category.has_children = with_children.include?(category[:id])
+      category.has_children = subcategory_count.key?(category[:id])
+
+      category.subcategory_count = subcategory_count[category[:id]] if category.has_children
```

Comment:
It feels like this needs a serializer spec somewhere to make sure the subcategory count is included and correct.